Add the visualizer to CI (and fix regressions)

Because the visualizer is not in CI, there is a danger of regression. We have already seen this with the subtopology work: https://github.com/nasa/fprime/issues/3868. The same thing can happen with respect to the XML tools: https://github.com/nasa/fprime/issues/3938.

Also, I suspect a regression may have happened already. When I run the visualizer in `Ref/Top`, I get this:
```
Built target refresh_cache
[ERROR] Context must be a Path
```
Possibly I am doing something wrong, but this looks like anomalous behavior.

--- a/src/fprime/fpp/cli.py
+++ b/src/fprime/fpp/cli.py
@@ -36,30 +36,6 @@ def run_fpp_check(
         args=({}, ["-u", parsed.unconnected] if parsed.unconnected else []),
     )
 
-
-def run_fpp_to_xml(
-    build: "Build",
-    parsed: argparse.Namespace,
-    _: Dict[str, str],
-    __: Dict[str, str],
-    ___: List[str],
-):
-    """Run the fpp-to-xml utility
-
-    Args:
-        build: build directory output
-        parsed: parsed input arguments
-        _: unused cmake_args
-        __: unused make_args
-        ___: unused pass-through arguments
-    """
-    FppUtility("fpp-to-xml").execute(
-        build,
-        parsed.path,
-        args=({}, ["--directory", parsed.directory] if parsed.directory else []),
-    )
-
-
 def run_fpp_to_dict(
     build: "Build",
     parsed: argparse.Namespace,
@@ -67,7 +43,7 @@ def run_fpp_to_dict(
     __: Dict[str, str],
     ___: List[str],
 ):
-    """Run the fpp-to-xml utility
+    """Run the fpp-to-dict utility
 
     Args:
         build: build directory output
@@ -112,12 +88,6 @@ def add_fpp_parsers(
         "-u", "--unconnected", default=None, help="write unconnected ports to file"
     )
 
-    fpp_to_xml_parser = subparsers.add_parser(
-        "fpp-to-xml", help="Run fpp-to-xml utility", parents=[common], add_help=False
-    )
-    fpp_to_xml_group = fpp_to_xml_parser.add_argument_group("fpp-to-xml arguments")
-    fpp_to_xml_group.add_argument("-d", "--directory", help="Output directory")
-
     fpp_to_dict_parser = subparsers.add_parser(
         "fpp-to-dict", help="Run fpp-to-dict utility", parents=[common], add_help=False
     )
@@ -127,10 +97,8 @@ def add_fpp_parsers(
 
     return {
         "fpp-check": run_fpp_check,
-        "fpp-to-xml": run_fpp_to_xml,
         "fpp-to-dict": run_fpp_to_dict,
     }, {
         "fpp-check": check_parser,
-        "fpp-to-xml": fpp_to_xml_parser,
         "fpp-to-dict": fpp_to_dict_parser,
     }

--- a/src/fprime/fpp/cli.py
+++ b/src/fprime/fpp/cli.py
@@ -1,4 +1,4 @@
-""" fprime.fpp.cli: FPP command line targets
+"""fprime.fpp.cli: FPP command line targets
 
 Processing and command line functions for FPP tools wrappers in fprime-util.
 
@@ -35,6 +35,7 @@ def run_fpp_check(
         parsed.path,
         args=({}, ["-u", parsed.unconnected] if parsed.unconnected else []),
     )
+
 
 def run_fpp_to_dict(
     build: "Build",

--- a/src/fprime/util/cli.py
+++ b/src/fprime/util/cli.py
@@ -324,7 +324,8 @@ def parse_args(args):
     common_parser.add_argument(
         "-p",
         "--path",
-        default=os.getcwd(),
+        default=Path.cwd(),
+        type=Path,
         help="F prime directory to operate on. Default: cwd, %(default)s.",
     )
     common_parser.add_argument(

--- a/src/fprime/util/commands.py
+++ b/src/fprime/util/commands.py
@@ -228,23 +228,12 @@ def run_version_check(
     print(f"Pip version: {pip.__version__}")
 
     print("Pip packages:")
-    # Used to print fprime-fpp-* versions together if they are all the same to de-clutter the output
-    fpp_packages = {}
     for tool in FPRIME_PIP_PACKAGES:
         try:
             version = pkg_resources.get_distribution(tool).version
-            if tool.startswith("fprime-fpp-"):
-                fpp_packages[tool] = version
-            else:
-                print(f"    {tool}=={version}")
+            print(f"    {tool}=={version}")
         except (OSError, VersionException, pkg_resources.DistributionNotFound) as exc:
             print(f"[WARNING] {exc}")
-    if fpp_packages:
-        if len(set(fpp_packages.values())) == 1:
-            print(f"    fprime-fpp-*=={list(fpp_packages.values())[0]}")
-        else:
-            for tool, version in fpp_packages.items():
-                print(f"    {tool}=={version}")
 
     try:
         out = (

--- a/src/fprime/util/versioning.py
+++ b/src/fprime/util/versioning.py
@@ -3,7 +3,7 @@
 import argparse
 import sys
 from pathlib import Path
-
+import pkg_resources
 
 class VersionException(Exception):
     pass
@@ -12,17 +12,7 @@ class VersionException(Exception):
 FPRIME_PIP_PACKAGES = [
     "fprime-tools",
     "fprime-gds",
-    "fprime-fpp-to-xml",
-    "fprime-fpp-to-json",
-    "fprime-fpp-to-cpp",
-    "fprime-fpp-syntax",
-    "fprime-fpp-locate-uses",
-    "fprime-fpp-locate-defs",
-    "fprime-fpp-from-xml",
-    "fprime-fpp-format",
-    "fprime-fpp-filenames",
-    "fprime-fpp-depend",
-    "fprime-fpp-check",
+    "fprime-fpp",
 ]
 
 

--- a/src/fprime/util/versioning.py
+++ b/src/fprime/util/versioning.py
@@ -1,9 +1,10 @@
-""" FPP tools to requirements file version check """
+"""FPP tools to requirements file version check"""
 
 import argparse
 import sys
 from pathlib import Path
 import pkg_resources
+
 
 class VersionException(Exception):
     pass

--- a/test/fprime/fpp/test_common.py
+++ b/test/fprime/fpp/test_common.py
@@ -21,8 +21,8 @@ def mock_build(tmp_path):
 
 def test_init():
     """Test FppUtility initialization"""
-    utility = FppUtility("fpp-to-xml")
-    assert utility.utility == "fpp-to-xml"
+    utility = FppUtility("fpp-to-dict")
+    assert utility.utility == "fpp-to-dict"
     assert utility.imports_as_sources is True
 
     utility_no_imports = FppUtility("fpp-check", imports_as_sources=False)
@@ -64,7 +64,7 @@ def test_get_fpp_inputs_missing(mock_build, tmp_path):
 
 
 @patch("subprocess.run")
-@patch("shutil.which", return_value="/path/to/fpp-to-xml")
+@patch("shutil.which", return_value="/path/to/fpp-to-dict")
 def test_execute_imports_as_sources(mock_which, mock_run, mock_build, tmp_path):
     """Test execute with imports as sources"""
     # Setup
@@ -72,14 +72,14 @@ def test_execute_imports_as_sources(mock_which, mock_run, mock_build, tmp_path):
     (tmp_path / "cache" / "fpp-import-list").write_text("A.fpp;B.fpp;")
     (tmp_path / "cache" / "fpp-source-list").write_text("C.fpp;D.fpp;")
 
-    utility = FppUtility("fpp-to-xml", imports_as_sources=True)
+    utility = FppUtility("fpp-to-dict", imports_as_sources=True)
     utility.execute(mock_build, tmp_path, ({}, []))
 
     # Check call
     mock_run.assert_called_once()
     call_args = mock_run.call_args[0][0]
     assert call_args == [
-        "fpp-to-xml",
+        "fpp-to-dict",
         str(tmp_path / "locs.fpp"),
         "A.fpp",
         "B.fpp",

--- a/test/fprime/util/commands_unit_test.py
+++ b/test/fprime/util/commands_unit_test.py
@@ -24,9 +24,6 @@ class CommandsTestCases(unittest.TestCase):
     def test_fpp_check_command(self):
         pass
 
-    def test_fpp_to_xml_command(self):
-        pass
-
     def test_visualize_command(self):
         pass
 

